### DISCUSSION TITLE
Custom Popup Functionality

### DISCUSSION THREAD

**Discussion by typenoob** [fyne-io/fyne#5340] [Q&A] [answered]
Hi, I want to make a popup which would be closed when tap on the backdrop but remain on the screen if tap inside of it. The `NewPopUp` or `NewModalPopUp` method does not satisfy my requirement perfectly. Is there any solution for it conveniently? Appreciate for any help!

**Comment by andydotxyz** [comment]
Please be more specific about what is not working - `PopUp` (not modal) should satisfy your requirement. There is also the

**Reply by typenoob** [reply]
> Please be more specific about what is not working - `PopUp` (not modal) should satisfy your requirement. There is also the

`Popup` also closes when I tap on its inside, but I want it to be only closed when I tap on the backdrop

**Reply by andydotxyz** [reply]
It shouldn't - the content of a popup would handle its own interactions... what did you put inside the popup?

**Comment by typenoob** [comment]
```
package main

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("demo")
	p := widget.NewPopUp(widget.NewLabel("demo"), w.Canvas())
	p.Show()
	w.Resize(fyne.NewSize(600, 800))
	w.ShowAndRun()
}
```

**Reply by typenoob** [reply]
I wrote the minimal example. When I clicked the content of the popup, it closed. Maybe label widget doesn't handle any interactions? But how I can make the whole popup widget handle all click interactions?

**Comment by andydotxyz** [comment]
Thanks for this, I guess all of the PopUps that I have used had tappable content which swallows this.

That would work for you too, but I guess this is an issue that should be resolved 
(we could make the background content of PopUp Tappable so it won't think it's a background tap)
I created an issue from the discussion to track this. I think the discussion should be marked as answered?

**Reply by andydotxyz** [reply, marked as answer]
This has been resolved on `develop` by @dweymouth so I think this should work as expected now.
